src/pages/index.tsx

```diff
@@ -69,8 +71,29 @@ function IndexPage() {
     },
     validate,
     onSubmit: (values) => {
-      console.log(values);
-      setFormSubmitted(true);
+      fetch("http://139.59.154.199:49160/api/v1/leads", {
+        method: "POST",
+        headers: {
+          Authorization: "secret_token",
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          name: values.username,
+          email: values.email,
+          consentsAccepted: values.acceptance,
+        }),
+      })
+        .then((response) => {
+          if (response.ok) {
+            setFormSubmitted(true);
+          } else if (response.status >= 500 && response.status < 600) {
```

Comment:
**nitpick:** Forgot to mention but you don't have to check this `500` status here. If `500` was thrown it would be automatically caught as error in `catch` below